graspologic/partition/leiden.py

```diff
@@ -8,210 +8,187 @@
-    starting_communities: Optional[Dict[str, int]] = None,
     extra_forced_iterations: int = 0,
-    resolution: float = 1.0,
-    randomness: float = 0.001,
-    use_modularity: bool = True,
+    resolution: Union[float, int] = 1.0,
+    randomness: Union[float, int] = 0.001,
     random_seed: Optional[int] = None,
-    is_weighted: Optional[bool] = None,
-    weight_default: float = 1.0,
-    check_directed: bool = True,
 ) -> None:
-    if starting_communities is not None and not isinstance(starting_communities, dict):
-        raise TypeError("starting_communities must be a dictionary")
-    if not isinstance(extra_forced_iterations, int):
-        raise TypeError("iterations must be an int")
-    if not isinstance(resolution, (int, float)):
-        raise TypeError("resolution must be a float")
-    if not isinstance(randomness, (int, float)):
-        raise TypeError("randomness must be a float")
-    if not isinstance(use_modularity, bool):
-        raise TypeError("use_modularity must be a bool")
-    if random_seed is not None and not isinstance(random_seed, int):
-        raise TypeError("random_seed must either be an int or None")
-    if is_weighted is not None and not isinstance(is_weighted, bool):
-        raise TypeError("is_weighted must either be a bool or None")
-    if not isinstance(weight_default, (int, float)):
-        raise TypeError("weight_default must be a float")
-    if not isinstance(check_directed, bool):
-        raise TypeError("check_directed must be a bool")
-
-    if extra_forced_iterations < 0:
-        raise ValueError("iterations must be a non negative integer")
-    if resolution <= 0:
-        raise ValueError("resolution must be a positive float")
-    if randomness <= 0:
-        raise ValueError("randomness must be a positive float")
-    if random_seed is not None and random_seed <= 0:
-        raise ValueError(
-            "random_seed must be a positive integer (the native PRNG implementation is"
-            " an unsigned 64 bit integer)"
-        )
+    check_argument(
+        extra_forced_iterations >= 0,
+        "extra_forced_iterations must be a non negative integer",
+    )
+    check_argument(resolution > 0, "resolution must be a positive float")
+    check_argument(randomness > 0, "randomness must be a positive float")
+    check_argument(
+        random_seed is None or random_seed > 0,
+        "random_seed must be a positive integer (the native PRNG implementation is"
+        " an unsigned 64 bit integer)",
+    )
 
 
+@beartype
 def leiden(
     graph: Union[
         List[Tuple[Any, Any, Union[int, float]]],
-        nx.Graph,
-        np.ndarray,
-        scipy.sparse.csr.csr_matrix,
+        GraphRepresentation,
     ],
-    starting_communities: Optional[Dict[str, int]] = None,
+    starting_communities: Optional[Dict[Any, int]] = None,
     extra_forced_iterations: int = 0,
-    resolution: float = 1.0,
-    randomness: float = 0.001,
+    resolution: Union[int, float] = 1.0,
```

Comment:
These need to be updated in the docstring

Reply:
👍